### PR TITLE
[network] Issue 6495

### PR DESCRIPTION
**Pull request by clinique**
Recreating PR for issue #6495 and adressing two potential NPE

### CODE REVIEW

**Comment by clinique** [comment]
Replaces PR #6631

File: bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/dhcp/DHCPPacketListenerServer.java [hunk 1]
```diff
@@ -70,7 +70,7 @@ protected void receivePacket(DHCPPacket request, @Nullable InetAddress udpRemote
 
         Byte dhcpMessageType = request.getDHCPMessageType();
 
-        if (dhcpMessageType != DHCPPacket.DHCPREQUEST) {
+        if (dhcpMessageType != null && dhcpMessageType != DHCPPacket.DHCPREQUEST) {
```
Comment (J-N-K) [inline bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/dhcp/DHCPPacketListenerServer.java]: What is the benefit of the null-check? `null` is not equal to `DHCPPacket.DHCPREQUEST` anyway.

Comment (clinique) [reply bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/dhcp/DHCPPacketListenerServer.java]: You're 100% right !

**Comment by martinvw** [comment]
@J-N-K can you take care of this one?

File: bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/handler/SpeedTestHandler.java [hunk 2]
```diff
@@ -132,7 +130,17 @@ public void onError(final @Nullable SpeedTestError testError, final @Nullable St
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMessage);
             freeRefreshTask();
             return;
-        } else if (SpeedTestError.SOCKET_TIMEOUT.equals(testError) || SpeedTestError.SOCKET_ERROR.equals(testError)
+        } else if (SpeedTestError.SOCKET_TIMEOUT.equals(testError)) {
+            timeouts--;
+            if (timeouts <= 0) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Max timeout count reached");
```
Comment (J-N-K) [inline bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/handler/SpeedTestHandler.java]: How can the user reset this counter? I'm thinking of e.g. a network connection problem which takes half a day or so. 

Or maybe a setting of `configuration.maxTimeouts=0` disables timeouts at all?

Comment (clinique) [reply bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/handler/SpeedTestHandler.java]: Disabling the thing is an option no ?

Comment (J-N-K) [reply bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/handler/SpeedTestHandler.java]: Sounds good. But what do you think of the timeout-counter disable?

**Comment by J-N-K** [comment]
We can add a disable later if needed.

**Pull request merged**